src/main/java/org/ohdsi/circe/cohortdefinition/CohortExpressionQueryBuilder.java

```diff
@@ -490,15 +490,21 @@ public String getCriteriaGroupQuery(CriteriaGroup group, String eventTable) {
       {
         if (group.type.toUpperCase().endsWith("LEAST"))
```

Comment:
Wouldn't strategy pattern fit here better? - now it's hard to read this nested `if`

Reply:
Actually, need to correct this:  you should only check for a group.count inside the AT_LEAST/AT_MOST if block, because group count could be zero (by default) but the group.type may be ALL or ANY which should result in a INNER JOIN, not a LEFT join. So, there will be 2 if statemetns inside the "AT_" check. I don't think it's hard to read tho.